libs/shared/shared_common.c

```diff
@@ -242,6 +242,10 @@ _duplicate_program_descriptor(
     program_type_descriptor_copy->header.version = EBPF_PROGRAM_TYPE_DESCRIPTOR_CURRENT_VERSION;
     program_type_descriptor_copy->header.size = EBPF_PROGRAM_TYPE_DESCRIPTOR_CURRENT_VERSION_SIZE;
 
+    // Initialize all pointers to NULL.
+    program_type_descriptor_copy->name = NULL;
+    program_type_descriptor_copy->context_descriptor = NULL;
```

Comment:
Line 235 does ebpf_allocate to get the space.
ebpf_allocate is cxplat_allocate(CXPLAT_POOL_FLAG_NON_PAGED, size, EBPF_POOL_TAG_DEFAULT).
cxplat_allocate zero-initializes all memory allocated, unless the CXPLAT_POOL_FLAG_UNINITIALIZED flag is passed, which this doesn't.  So why this change?

Reply:
There is a `memcpy` at line 241 that copies the original struct to new struct. That also copies the embedded pointers to the new struct, and that causes double free issues.

This change is resetting all the embedded pointers to NULL before proceeding.
Same pattern at other places in this PR.

Reply:
This is redundant initialization:
```
    program_type_descriptor_copy->name = NULL;
    program_type_descriptor_copy->name = cxplat_duplicate_string(program_type_descriptor->name);
```